lightning/src/onion_message/messenger.rs

```diff
@@ -243,6 +243,66 @@ impl OnionMessageRecipient {
 	}
 }
 
+/// A struct handling response to an [`OnionMessage`]
+pub struct Responder<'a, OMH: OnionMessageHandler, T: OnionMessageContents> {
+	messenger: &'a OMH,
+	pub reply_path: BlindedPath,
+	pub path_id: Option<[u8; 32]>,
```

Comment:
I'm not entirely sure if it is possible, but it would be nice to have some symmetry in the enum variants by having the `path_id` directly in both. Perhaps by storing `FormatArgs`? But you may run into some lifetime issues. Copying the path_id wouldn't be horrible, though.